java/system-test/system-test-suite/src/test/java/sleeper/systemtest/suite/testutil/PurgeQueueOnTestFailureExtension.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.systemtest.suite.testutil;
+
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import sleeper.configuration.properties.instance.InstanceProperty;
+import sleeper.systemtest.suite.dsl.ingest.SystemTestIngest;
+
+import java.util.function.Consumer;
+
+public class PurgeQueueOnTestFailureExtension implements AfterEachCallback {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PurgeQueueOnTestFailureExtension.class);
+    private final InstanceProperty queueProperty;
+    private final Consumer<InstanceProperty> purgeQueue;
+    private final Runnable waitFn;
+
+    PurgeQueueOnTestFailureExtension(InstanceProperty queueProperty, Consumer<InstanceProperty> purgeQueue, Runnable waitFn) {
+        this.queueProperty = queueProperty;
+        this.purgeQueue = purgeQueue;
+        this.waitFn = waitFn;
+    }
+
+    public static PurgeQueueOnTestFailureExtension withQueue(InstanceProperty queueProperty, SystemTestIngest ingest) {
+        return new PurgeQueueOnTestFailureExtension(queueProperty, ingest::purgeQueue, () -> {
+            try {
+                LOGGER.info("Waiting 60s for queue to purge");
+                Thread.sleep(60000L);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
```

Comment:
Catching an InterruptedException clears the thread's interrupted state, so the interrupt may not take effect. You can re-interrupt the thread with Thread.currentThread().interrupt() before you throw the RuntimeException, or throw the InterruptedException out of the Lambda by defining an interface that throws it.